substrate/client/transaction-pool/src/graph/validated_pool.rs

```diff
@@ -692,6 +693,11 @@ impl<B: ChainApi> ValidatedPool<B> {
 	/// to prevent them from entering the pool right away.
 	/// Note this is not the case for the dependent transactions - those may
 	/// still be valid so we want to be able to re-import them.
+	///
+	/// For every removed transaction an Invalid event is triggered.
+	///
+	/// Returns the list of actually removed transactions, which may include transactions dependent
+	/// on provided set.
 	pub fn remove_invalid(&self, hashes: &[ExtrinsicHash<B>]) -> Vec<TransactionFor<B>> {
```

Comment:
This looks like we should unify with `remove_subtree` below, its mostly the same thing.

Reply:
Yeah I was hesitating about this. Will do this.

Reply:
Done in: 4200ccc